service-proxy.js

```diff
@@ -172,6 +172,24 @@ function rateLimit(req, buildRes) {
     var isExitNode = self.isExitFor(req.serviceName);
     if (isExitNode) {
         self.rateLimiter.createServiceCounter(req.serviceName);
+        self.rateLimiter.createKillSwitchCounter(req.serviceName);
+    }
+
+    // apply kill switch safe guard first
+    if (isExitNode && self.rateLimiter.shouldKillSwitchService(req.serviceName)) {
```

Comment:
Good point about the test! I will add it. W.r.t. the name, I would like to use kill switch, since they are essentially doing the same thing: through manual or through rate limiting. :)